utError.js

```diff
@@ -86,6 +89,9 @@ module.exports = ({logFactory, logLevel}) => {
                     Object.defineProperty(error, 'name', {value: type, configurable: true, enumerable: false});
                     error.type = type;
                     error.message = interpolate(props.message, params.params);
+                    if (definition.transformer) {
+                        error = definition.transformer(error);

```

Comment:
why do you need to transform the error? Can you please provide an example

Reply:
The idea behind this change is that we are working on a temporary Solution Controller, based on ut framework that needs to have the same APIs as our current IB Java Solution Controller.
We need error transforming in order to have the same error structure:
`
{
    "error": {
        "message": "Login error: loginErrUser",
        "code": "allianz.login"
        // optionally more things
    }
}
`
Because this SC will be used by a mobile app and to avoid the need to parse and transform errors in the mobile app.

Reply:
I personally think it is not a good idea to patch standard behaviour for the sake of changing it to something else that doesn't improve anything. What is the difference between providing a `transformer` and changing the error in the code before returning it to the client?

Or instead of defining a transformer function just to wrap the error factory function like:

```js
Object.assign(
    this.errors,
    Object.entries(registerErrors(errors))
        .reduce((all, [type, fn]) => ({...all, [type]: (...params) => {
            const err = fn(...params);
            // do whatever
            return err;
        }}), {})
);
```

what is the difference?

Reply:
Another option would be to define just 1 error:

```json
{
    "solution.controller": "Solution controller error"
}

```

then in js code:

```js
import: {
    errorSolutionController
}

// ...

throw errorSolutionController({
    error: {
        code: 'x',
        message: 'y'
    }
});
```

and the resulting error will be in format:

```json
{
    "type": "solution.controller",
    "message": "Solution controller error",
    "error": {
        "code": "x",
        "message": "x"
    }
}
```

Reply:
We need to align the approach to how things are defined in the openapi/swagger. I am not sure if the per error transform is the proper approach, as there might be various errors, including ones coming from the framework. I will try to do some research.